### PR TITLE
Cross project proposal navigation tests

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

This PR addresses the missing tests for cross-project navigation.

# Changes

- Added a test to verify that navigation from Nns to Sns works (Note: Testing navigation from Sns to Nns is more complex due to the need for mocking two different proposal type components).
- Added a test to verify that navigation between different Sns instances works.

# Tests

Yes.

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary.

### CODE REVIEW

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts [hunk 1]
```diff
@@ -399,17 +395,92 @@ describe("SnsProposalDetail", () => {
...
+    });
+    fakeSnsGovernanceApi.addProposalWith({
+      identity: mockIdentity,
+      rootCanisterId: rootCanisterId2,
+      id: [{ id: 2n }],
+    });
+
+    const { container } = render(SnsProposalDetail, {
+      props: {
+        // set the proposal with id=2 to be in the middle of the list
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:412]: The line below has `"1"` so I'm confused what `id=2` in the comment refers to here?

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts [hunk 2]
```diff
@@ -428,16 +499,20 @@ describe("SnsProposalDetail", () => {
...
-    expect(await navigationPo.getNextButtonProposalUniverse()).toBe(
-      rootCanisterId3.toText()
-    );
+
+    await navigationPo.clickPrevious();
+    expect(get(page)).toEqual({
+      data: {
+        actionable: "",
+        proposal: "1",
+        universe: "f7crg-kabae",
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:510]: Can you declare constants for the universe ID strings so it's clear how they related to the rootCanisterIds?

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts [hunk 3]
```diff
@@ -428,16 +499,20 @@ describe("SnsProposalDetail", () => {
...
+      },
+      route: {
+        id: "/(app)/proposal/",
+      },
+    });
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:515]: You have a number of tests with quite large, very similar, setup. Do you think it could improve readability to extra this setup to a utility function in this file (it's not generic enough to put it in a separate utils file)?

File: frontend/src/tests/lib/components/proposal-detail/NnsProposal.spec.ts
Comment (dskloetd) [inline frontend/src/tests/lib/components/proposal-detail/NnsProposal.spec.ts:1]: What do you think about removing the `data-` attributes that were added to test the buttons.
Aren't they redundant now that you test the actual navigation?

Comment (mstrasinskis) [reply frontend/src/tests/lib/components/proposal-detail/NnsProposal.spec.ts:1]: Yes, they were here as a workaround for the actual navigation test. It also seems to me that we might be abusing our unit tests a bit, and they are no longer strictly unit tests :)

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts (hunk 3 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:474]: Good point. They were moved to [the `beforeEach` block.](https://github.com/dfinity/nns-dapp/pull/5018/files#diff-a6a3df92762c57ff017db82a006d8fd5ae86bfde786851949cc7e2e5beed6456R359) I find it too specific to have separate utility functions for this.

Comment (dskloetd) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:478]: I was hoping to see within the same test why the expected universe and proposal ID are the correct ones. By moving the test data out of the test, you made it harder to see rather than easier.

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts]: Done.

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts (hunk 3 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:465]: Refactored. Could you re-check it?

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts [hunk 4]
```diff
@@ -355,89 +356,113 @@ describe("SnsProposalDetail", () => {
...
-      {
-        rootCanisterId: rootCanisterId3,
+  describe("cross-project navigation", () => {
+    const principal1 = principal(11);
+    const principal2 = principal(22);
+
+    const mockCommittedSnsProjectsWithVotableProposal = (
+      props: Array<{ rootCanisterId: Principal; proposalId: bigint }>
+    ) => {
+      const snsProjects = props.map(({ rootCanisterId }) => ({
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:366]: This might include duplicate projects, right?

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts (hunk 3 above)
Comment (dskloetd) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:465]: Yes, this is much easier to read. Thanks!

File: frontend/src/tests/lib/components/proposal-detail/NnsProposal.spec.ts
Comment (dskloetd) [reply frontend/src/tests/lib/components/proposal-detail/NnsProposal.spec.ts:1]: So can you remove them in this PR since they are no longer used?

**Review by dskloetd — APPROVED** [review]

File: frontend/src/tests/lib/components/proposal-detail/NnsProposal.spec.ts
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/proposal-detail/NnsProposal.spec.ts:1]: Done.

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts (hunk 4 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:366]: Correct. Do you see any issues with that?

Comment (dskloetd) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:366]: I haven't checked but I wouldn't necessarily expect `setSnsProjects` to work with duplicate projects.

**Review by dskloetd — APPROVED** [review]

File: frontend/src/lib/components/proposal-detail/ProposalNavigation.svelte [hunk 5]
```diff
@@ -69,8 +69,6 @@
     on:click={selectPrevious}
     class:hidden={isNullish(previousId)}
     data-tid="proposal-nav-previous"
-    data-test-proposal-id={previousId?.proposalId.toString() ?? ""}
```
Comment (dskloetd) [inline frontend/src/lib/components/proposal-detail/ProposalNavigation.svelte:72]: Judging from the failing tests, I guess you need to keep this one for now. But the one you added more recently, I expect you can remove.

**Review by dskloetd — APPROVED** [review]

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts (hunk 4 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:366]: There is `resetSnsProjects` called [before every test](https://github.com/dfinity/nns-dapp/pull/5018/files#diff-a6a3df92762c57ff017db82a006d8fd5ae86bfde786851949cc7e2e5beed6456R62), so should be reliable. Or did I miss something?

Comment (dskloetd) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:366]: I'm talking about this `snsProjects` variable right here having the same `rootCanisterId` multiple times during a single call into this function.
The tests you made happen to have only 1 proposal ID per project. But that's not very realistic. You might make a more realistic test with multiple proposals per project and then this array variable would have duplicates, right?

Comment (mstrasinskis) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts]: Thanks, the mock function was extended to support multiple proposals.

**Review by dskloetd — APPROVED** [review]
